scripts/unify

```diff
@@ -1,36 +1,59 @@
-'use strict';
+#!/usr/bin/env node
+
+/**
+ * Copyright 2017, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
-// Copyright 2016, Google, Inc.
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//    http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+'use strict';
 
 const fs = require(`fs`);
 const path = require(`path`);
 
 const PROJECT_ROOT = path.join(__dirname, `..`);
 
-// Get subdirectories with a `package.json` file
-const directories = fs.readdirSync(PROJECT_ROOT)
-                      .filter((dir) => fs.existsSync(path.join(PROJECT_ROOT, dir, `package.json`)));
-
 // Dedupe package.json dependencies
 // WARNING: This will fail if two different versions of the same package are required.
-let pkgSet = {};
-let pkgJson;
-directories.forEach((dir) => {
-  pkgJson = JSON.parse(fs.readFileSync(path.join(PROJECT_ROOT, dir, `package.json`)));
-  Object.assign(pkgSet, pkgJson.dependencies);
-});
+const pkgSet = {};
+const pkgDevSet = {};
+
+function getDeps (directory) {
+  // Get subdirectories with a `package.json` file
+  const directories = fs.readdirSync(directory)
+    .filter((dir) => fs.existsSync(path.join(directory, dir, `package.json`)));
 
-// Update root-level package.json (by shelling to yarn)
-const spawn = require('child_process').spawn;
-spawn(`yarn`, [`add`].concat(Object.keys(pkgSet).map(pkg => `${pkg}@${pkgSet[pkg]}`)));
+  let pkgJson;
+  directories.forEach((dir) => {
+    pkgJson = JSON.parse(fs.readFileSync(path.join(directory, dir, `package.json`)));
+    Object.assign(pkgSet, pkgJson.dependencies);
+    Object.assign(pkgDevSet, pkgJson.devDependencies);
+  });
+}
+
+getDeps(PROJECT_ROOT);
+
+console.log(JSON.stringify(pkgSet, null, 2));
```

Comment:
Is this debug code, or is it supposed to be here? (If it is supposed to be here, perhaps we should include a message explaining what this JSON is - e.g. `Unified package.json contents`)